casper/src/main/scala/coop/rchain/casper/ReportingCasper.scala

```diff
@@ -175,10 +153,25 @@ object ReportingCasper {
                       case Left(_)  => Seq.empty[Seq[ReportingEvent]]
                       case Right(s) => s
                     }
-                  } yield (term, r)
-                })
-          _ <- EitherT.right[ReplayFailure](runtime.createCheckpoint)
-        } yield res).value
+                  } yield DeployReportResult(term, r)
+                }
+          sysRes <- systemDeploys.toList.traverse { term =>
+                     for {
+                       _ <- Log[F].info(s"Replay system deploy ${term.systemDeploy}")
+                       rd <- runtime
+                              .replaySystemDeployE(blockData)(term)
+                              .semiflatMap(_ => runtime.getReport)
+                              .value
+                       r = rd match {
+                         case Left(_)  => Seq.empty[Seq[ReportingEvent]]
+                         case Right(s) => s
+                       }
+                     } yield SystemDeployReportResult(term.systemDeploy, r)
+                   }
+
+          checkPoint <- runtime.createCheckpoint
```

Comment:
`createCheckpoint` here would store the items again which is not neccessary at all.  All we need here is the proof while the history current implementation is difficult to change the storing part we would leave that into another pr.